### ISSUE TITLE
Parsing publisher list

### ISSUE THREAD

**Issue by Rekyt** [Rekyt/rromeo#13]
For the moment `parse_answer()` only parses well list of journals.
However, I'm trying to implement `rr_romeo_colour()` to query by RoMEO colour. This query returns a publisher list that `parse_answer()` doesn't parse.

So we should modify it so that it recognizes whether the query contains lists of journals or lists of publishers.

See the WIP code here 4963e489e2842cc71f5fc5ce0fd597ececbd5b00

**Comment by Bisaloo** [comment]
The structure of the XML file is really different in this case. We might be better off writing a different function.

Because otherwise, we will probably end up with something like:

```r
parse_answer <- function(...) {
  if (publisher) {
    ...
  } else if (journal) {
    ...
  }
}
```

**Comment by Bisaloo** [comment]
Hum, I'm not sure anymore. Now that we test on `outcome` (a9fd2821fc67c7f957b526115f9086be221e1d3a), it's not that difficult to do what I mentioned earlier. This would also reduce a little code duplication (but very little).

Basically, this is just a stylistic preference. So I don't know...

**Comment by Rekyt** [comment]
Can't we have both like:

```r
parse_answer <- function(...) {
  if (publisher) {
    parse_publisher()
  } else if (journal) {
    parse_journal()
  }
}
```
To get the best of the two worlds?

**Comment by Rekyt** [comment]
Actually fixed by f12432d10bf
